swaggen/src/main/java/generator/PathGenerator.java

```diff
@@ -54,9 +54,57 @@
 	 * @param annotation the annotation
 	 * @return the path
 	 */
-	private static Map<RequestMethod, Endpoint> generatePathFromAnnotation(SwaggerGen annotation) {
-		Map<RequestMethod, Endpoint> endpointMap = new HashMap<>();
-		endpointMap.put(RequestMethod.valueOf(annotation.method()), EndpointFactory.createEndpoint(annotation));
-		return endpointMap;
+	private static SwaggerEndpoint generatePathFromAnnotation(SwaggerGen annotation) {
+		SwaggerEndpoint endpoint = new SwaggerEndpoint();
+		endpoint.addEndpoint(RequestMethod.valueOf(annotation.method()), EndpointFactory.createEndpoint(annotation));
+		return endpoint;
+	}
+
+	/**
+	 * Checks if URL is already in the existing map, and if it is, appends the swagger
+	 * endpoints to the URL
+	 * Case: multiple classes with the same endpoint
+	 * @param existingPaths Map of existing Swagger Endpoints
+	 * @param classPaths Map of new Swagger Endpoints in the class
+	 */
+	private static Map<PathUri, SwaggerEndpoint> checkClassEndpoints(Map<PathUri, SwaggerEndpoint> existingPaths, Map<PathUri, SwaggerEndpoint> classPaths) {

```

Comment:
This method and the one below it look correct now. Sonarqube is still going to be annoyed with the triply-nested statements, but that is an issue for later. Good work